toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1355,23 +1445,33 @@ func (g *PkgGraph) fixPrebuiltSRPMsCycle(trimmedCycle []*PkgNode) (err error) {
 
 			logger.Log.Debugf("Adding a 'PreBuilt' node '%s' with id %d.", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
-			parentNodes := g.To(currentNode.ID())
-			for parentNodes.Next() {
-				parentNode := parentNodes.Node().(*PkgNode)
-				if parentNode.Type == TypeLocalBuild && parentNode.SrpmPath == previousNode.SrpmPath {
-					g.RemoveEdge(parentNode.ID(), currentNode.ID())
-
-					err = g.AddEdge(parentNode, preBuiltNode)
-					if err != nil {
-						logger.Log.Errorf("Adding edge failed for %v -> %v", parentNode, preBuiltNode)
-						return
-					}
-				}
-			}
+			g.replaceCurrentNodeWithPrebuiltNode(currentNode, preBuiltNode, previousNode)
+			logger.Log.Infof("Cycle fixed using prebuilt node: %s with id %d", preBuiltNode.FriendlyName(), preBuiltNode.ID())
 
 			return
 		}
-
+		if resolveCyclesFromUpstream {
+			isAvailableUpstream := false
+			var clonerErr error
+			isAvailableUpstream, clonerErr = isSRPMAvailableUpstream(g, nil, currentNode.SrpmPath, ignoreVersionToResolveSelfDep, cloner)
+			/*clonerErr is not fatal. gives error if rpm is not found*/
+			if clonerErr != nil {
+				logger.Log.Debugf("Error while checking if SRPM is available upstream: %v", clonerErr)
+			}
+			if buildToRunEdge && isAvailableUpstream {
+				//Mark node as unresolved remote to be fetched by pkgfetcher
+				logger.Log.Debugf("Cycle contains SRPM %s whose rpms are all available in PMC. MARK this node as remote unresolved", currentNode.SrpmPath)
```

Comment:
With the two following debug comments, I think we can drop this one. The other two seem to be providing the same info.

Reply:
Done